js/background/auxiliary.js

```diff
@@ -12,26 +12,32 @@ setInterval(() => {
  */
 chrome.runtime.onMessage.addListener((req, sender, sendResponse) => {
     if (req.action == 'request') {
-        const { data: { url } } = req;
-        if (REQUEST_CACHE[url]) {
-            sendResponse(REQUEST_CACHE[url]);
-            return true;
-        } 
+
         let res = { status: 'error', data: null };
-        fetch(url)
-            .then((data) => { return data.json(); })
-            .then((data) => {
-                res.data = data;
-                res.status = 'OK';
-                REQUEST_CACHE[url] = res;
-                sendResponse(res);
-            })
-            .catch((error) => {
-                res.data = error;
-                res.status = 'error';
-                sendResponse(res);
-                console.error('SURVOL - Fetching error', error);
-            });
+
+        // if the request is cached
+        if (REQUEST_CACHE[url]) {
```

Comment:
Does this need to be req.data.url?

Reply:
It does thank you I tried to rush things, should have tested :/ fixed #4a1d9f0